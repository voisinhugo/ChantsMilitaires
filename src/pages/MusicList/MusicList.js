import React from 'react';
import {FlatList, StyleSheet, SafeAreaView} from 'react-native';
import songs from '../../data';
import {MusicItem} from './components';
import theme from '../../theme';

const styles = StyleSheet.create({
  container: {
    backgroundColor: theme.color.background,
  },
  musicList: {
    backgroundColor: theme.color.background,
  },
});

export const MusicList = ({navigation}) => (
  <SafeAreaView style={styles.container}>
    <FlatList
      style={styles.musicList}
      data={Object.values(songs).sort((a, b) => a.title > b.title)}
      renderItem={({item}) => <MusicItem item={item} navigation={navigation} />}
      keyExtractor={item => item.id.toString()}
    />
  </SafeAreaView>
);
