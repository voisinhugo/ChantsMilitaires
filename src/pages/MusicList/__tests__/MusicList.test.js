import React from 'react';
import {render} from 'react-native-testing-library';
import {MusicList} from '../MusicList';
import {getPropsWithNavigation} from '../../../lib/tests/helpers';

jest.mock('../../../data/index', () => ({
  '0': {id: 0, title: 'toto'},
  '1': {id: 1, title: 'tata'},
}));

const navigation = getPropsWithNavigation();

describe('MusicItem', () => {
  it('should render', () => {
    const musicItem = render(<MusicList navigation={navigation} />);

    expect(musicItem).toMatchSnapshot();
  });
});
