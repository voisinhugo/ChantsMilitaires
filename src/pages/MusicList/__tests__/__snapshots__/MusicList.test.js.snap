// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicItem should render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#272727",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 1,
          "title": "tata",
        },
        Object {
          "id": 0,
          "title": "toto",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#272727",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#272727",
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 20,
                "marginLeft": 16,
                "marginVertical": 8,
              }
            }
          >
            tata
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#272727",
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 20,
                "marginLeft": 16,
                "marginVertical": 8,
              }
            }
          >
            toto
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
